perfkitbenchmarker/aws/aws_virtual_machine.py

```diff
@@ -230,7 +230,25 @@ def _Delete(self):
         'terminate-instances',
         '--region=%s' % self.region,
         '--instance-ids=%s' % self.id]
-    vm_util.IssueRetryableCommand(delete_cmd)
+    vm_util.IssueCommand(delete_cmd)
+
+  def _Exists(self):
+    """Returns true if the VM exists."""
+    describe_cmd = util.AWS_PREFIX + [
+        'ec2',
+        'describe-instances',
+        '--region=%s' % self.region,
+        '--filter=Name=instance-id,Values=%s' % self.id]
+    stdout, _ = vm_util.IssueRetryableCommand(describe_cmd)
+    response = json.loads(stdout)
+    reservations = response['Reservations']
+    if not reservations:
+      return False
+    status = reservations[0]['Instances'][0]['State']['Name']
+    if status in ['shutting-down', 'terminated']:
```

Comment:
As above: consider module level constants for these. Also consider asserting that the status is a known value, in case of API changes.